### PR TITLE
Update cloudtrail parser

### LINKED ISSUE #4182: AWS CloudTrail parser
**Description of problem:**

The aws cloud trail log parser does not work with current cloudtrail data.

**Source data:**

https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-log-file-examples.html#error-code-and-error-message

**Plaso version:**

For example 20220724

### PR DESCRIPTION
**Pull request by alexgoedeke**
## One line description of pull request

add current log format for aws cloudtrail parser.

## Description:

**Related issue (if applicable):** fixes #4182 

## Notes:
All contributions to Plaso undergo [code
review](https://github.com/log2timeline/l2tdocs/blob/main/process/Code%20review%20process.asciidoc). This makes sure
that the code has appropriate test coverage and conforms to the Plaso [style
guide](https://plaso.readthedocs.io/en/latest/sources/developer/Style-guide.html).

One of the maintainers will examine your code, and may request changes. Check off the items below in
order, and then a maintainer will review your code.

## Checklist:
  - [x] Automated checks (Travis, Codecov, Codefactor )pass
  - [x] No new [new dependencies](https://plaso.readthedocs.io/en/latest/sources/developer/Adding-a-new-dependency.html) are required or l2tdevtools has been updated
  - [ ] Reviewer assigned

### CODE REVIEW

**Comment by jleaniz** [comment]
@alexgoedeke Thanks for the contribution! At first glance, it seems like the test data you are using here differs from the test data in the original parser (e.g. not in JSONL format) and more closely matches what is provided in  https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-log-file-examples.html - looks like the format is in fact JSON and not JSONL. @jonathan-greig

**Comment by jonathan-greig** [comment]
Hi Alexander, thanks for this PR but Juan is correct that the Cloudtrail log parser is designed for logs saved in JSON-L format and your changes seem to change the parser to use JSON. [dftimewolf](https://github.com/log2timeline/dftimewolf) with the `aws_logging_collect` can be used to save cloudtrail logs in JSON-L format.

**Comment by jleaniz** [comment]
@alexgoedeke I think it would make sense to have this parser in addition to the existing one, rather than as a replacement. This new parser would seem to work on log data directly exported from AWS Cloudtrail. Would you be willing to adjust this PR accordingly?

**Comment by alexgoedeke** [comment]
@jleaniz Yes, this makes sense. I adjusted the PR accordingly.

**Comment by joachimmetz** [comment]
@alexgoedeke how is this parser different from https://github.com/log2timeline/plaso/blob/main/plaso/parsers/jsonl_plugins/aws_cloudtrail_log.py ?

File: plaso/parsers/aws_cloudtrail_log.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
+# -*- coding: utf-8 -*-
+"""JSON-L parser plugin for AWS CloudTrail log files.
```
Comment (joachimmetz) [inline plaso/parsers/aws_cloudtrail_log.py:2]: Per comment thread, is this a JSON or JSON-L parser?

File: plaso/parsers/aws_cloudtrail_log.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+"""
+
+import json
+import os
+
+
+from dfdatetime import semantic_time as dfdatetime_semantic_time
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+from plaso.containers import events
```
Comment (joachimmetz) [inline plaso/parsers/aws_cloudtrail_log.py:16]: style nit: order alphabetically

File: plaso/parsers/aws_cloudtrail_log.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+from plaso.containers import time_events
+from plaso.lib import definitions
+from plaso.parsers import interface
+
+
+class AWSCloudTrailEventData(events.EventData):
+  """AWS CloudTrail log event data.
+
+  Attributes:
+    eventTime
```
Comment (joachimmetz) [inline plaso/parsers/aws_cloudtrail_log.py:28]: style nit: missing docstrings and docstring does not match the actual attribute names

File: plaso/parsers/aws_cloudtrail_log.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+      'vpcEndpointId': 'vpc_endpoint_identifier',
+      'eventCategory': 'event_category',
+      'addendum': 'addendum',
+      'sessionCredentialFromConsole': 'session_credential_from_console',
+      'edgeDeviceDetails': 'edge_device_details',
+      'tlsDetails': 'tls_details'
+  }
+
+  _JSON_PROPERTIES = {
+    'userIdentity': 'user_identity',
```
Comment (joachimmetz) [inline plaso/parsers/aws_cloudtrail_log.py:131]: Use 4 spaces as continuation indentation (repeat this in other places)

File: tests/parsers/aws_cloudtrail_log.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    number_of_warnings = storage_writer.GetNumberOfAttributeContainers(
+        'recovery_warning')
+    self.assertEqual(number_of_warnings, 0)
+
+    events = list(storage_writer.GetEvents())
+
+    # pylint: disable=line-too-long
+
+    expected_event_values = {
+        "event_version": "1.0",
```
Comment (joachimmetz) [inline tests/parsers/aws_cloudtrail_log.py:37]: style nit: plaso codebase uses single quotes

File: tests/parsers/aws_cloudtrail_log.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        "aws_region": "us-east-1",
+        "event_identifier": "EXAMPLE8-9621-4d00-b913-beca2EXAMPLE",
+        "event_type": "AwsCloudTrailInsight",
+        "recipient_account_identifier": "123456789012",
+        "shared_event_identifier": "EXAMPLE2-1729-42f1-b735-5d8c0EXAMPLE",
+        "insight_details": '{"state": "Start", "eventSource": "ssm.amazonaws.com", "eventName": "UpdateInstanceInformation", "insightType": "ApiCallRateInsight", "insightContext": {"statistics": {"baseline": {"average": 85.4202380952}, "insight": {"average": 664}}}}',
+        "event_category": "Insight"
+    }
+    self.CheckEventValues(storage_writer, events[7], expected_event_values)
+
```
Comment (joachimmetz) [inline tests/parsers/aws_cloudtrail_log.py:76]: style nit: missing empty line

**Review by joachimmetz — CHANGES_REQUESTED** [review]
Unclear how this parser differs from the current JSON-L plugin

**Comment by joachimmetz** [comment]
@alexgoedeke what is the status of this? Should I close this PR for now and reopen when you have the time work on it?

**Comment by joachimmetz** [comment]
per previous comment closing PR for now

**Pull request closed without merging**